packages/components/src/internal/components/project/ProjectDataTypeSelections.tsx

```diff
@@ -60,35 +81,50 @@ export const ProjectDataTypeSelections: FC<Props> = memo(props => {
             setDirty(false);
             onSuccess(true);
         } catch (e) {
-            setError(resolveErrorMessage(e) ?? 'Failed to update project settings');
+            setSaveError(resolveErrorMessage(e) ?? 'Failed to update project settings');
         } finally {
             setIsSaving(false);
         }
-    }, [isSaving, onSuccess, project, dataTypeExclusion]);
+    }, [isSaving, dataTypeExclusion, api, project?.path, onSuccess]);
 
     return (
         <div className="panel panel-default">
-            <div className="panel-heading">Data in Project</div>
+            <div className="panel-heading">{panelTitle}</div>
             <div className="panel-body">
                 <div className="form-horizontal">
                     {error && <Alert>{error}</Alert>}
-                    <div className="bottom-spacing">Select the types of data that will be used in this project.</div>
-                    <Row>
-                        {entityDataTypes?.map((entityDataType, index) => {
-                            return (
-                                <Col key={index} xs={12} md={4} className="bottom-spacing">
-                                    <DataTypeSelector
-                                        entityDataType={entityDataType}
-                                        updateUncheckedTypes={updateDataTypeExclusions_}
-                                        uncheckedEntitiesDB={
-                                            disabledTypesMap?.[entityDataType.projectConfigurableDataType]
-                                        }
-                                        isNewFolder={!project}
-                                    />
-                                </Col>
-                            );
-                        })}
-                    </Row>
+                    {saveError && <Alert>{saveError}</Alert>}
+                    <div className="bottom-spacing">{panelDescription}</div>
+                    {!loaded && <LoadingSpinner />}
+                    <div className="row">
+                        {loaded &&
+                            entityDataTypes?.map(entityDataType => {
+                                const uncheckedDataTypes =
```

Comment:
maybe add a comment to explain what hidden and unchecked each mean...

Reply:
done